File: epi_judge_python/tree_level_order.py
from test_framework import generic_test

from binary_tree_node import BinaryTreeNode


def binary_tree_depth_order(tree: BinaryTreeNode):
    result = []
    if tree is None:
        return result
    q1 = [tree]
    while q1:
        result.append([node.data for node in q1])
        q1 = [child for node in q1 for child in (node.left, node.right) if child is not None]
    return result


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("tree_level_order.py",
                                       "tree_level_order.tsv",
                                       binary_tree_depth_order))

File: epi_judge_java_solutions/epi/TreeLevelOrder.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class TreeLevelOrder {
  @EpiTest(testDataFile = "tree_level_order.tsv")

  public static List<List<Integer>>
  binaryTreeDepthOrder(BinaryTreeNode<Integer> tree) {

    List<List<Integer>> result = new ArrayList<>();
    if (tree == null) {
      return result;
    }

    List<BinaryTreeNode<Integer>> currDepthNodes = List.of(tree);
    while (!currDepthNodes.isEmpty()) {
      result.add(currDepthNodes.stream()
                     .map(curr -> curr.data)
                     .collect(Collectors.toList()));
      currDepthNodes = currDepthNodes.stream()
                           .map(curr -> Arrays.asList(curr.left, curr.right))
                           .flatMap(List::stream)
                           .filter(child -> child != null)
                           .collect(Collectors.toList());
    }
    return result;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "TreeLevelOrder.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
